# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build sdl2-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists flac-build && REQUIRES="$REQUIRES flac-build"
pkg-config --exists libvorbis-build && REQUIRES="$REQUIRES libvorbis-build"

# configure (ac-2.69; lt-2.2.6)
libtoolize -c -f -i; cat acinclude/* > aclocal.m4; autoconf
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install build/.libs/playmus $DESTDIR/x11/bin/playmus2
install build/.libs/playwave $DESTDIR/x11/bin/playwave2
install -d $DESTDIR/x11/include/SDL2
install -m 644 SDL_mixer.h $DESTDIR/x11/include/SDL2
install -d $DESTDIR/x11/lib
install build/.libs/libSDL2_mixer-2.0.so.0 $DESTDIR/x11/lib
ln -sf libSDL2_mixer-2.0.so.0 $DESTDIR/x11/lib/libSDL2_mixer.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 SDL2_mixer.pc $DESTDIR/x11/lib/pkgconfig

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/sdl2_mixer-build.pc << EOF
Name: SDL2_mixer
Version: 2.0.4
Description: Audio mixer library for SDL
Requires: $REQUIRES

devel=\\
/x11/include/SDL2/SDL_mixer.h \\
/x11/lib/libSDL2_mixer.so \\
/x11/lib/pkgconfig/SDL2_mixer.pc

exec=\\
/x11/bin/playmus2 \\
/x11/bin/playwave2 \\
/x11/lib/libSDL2_mixer-2.0.so.0
EOF

exit

# patch
--- SDL2_mixer-2.0.4.orig/configure.in
+++ SDL2_mixer-2.0.4/configure.in
@@ -1,6 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(README.txt)
 AC_CONFIG_AUX_DIR(build-scripts)
+AC_CONFIG_MACRO_DIR([acinclude])
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
# source
https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-2.0.4.tar.gz
